src/main/clojure/clara/rules/engine.cljc

```diff
@@ -337,17 +337,17 @@
   ILeftActivate
   (left-activate [node join-bindings tokens memory transport listener]
 
+    ;; Provide listeners information on all left-retract calls for passivity,
```

Comment:
Just a minor thing, but I think it is more important than just preserving old behavior. I think it is really useful behavior to be able to listen for this activity in the ProductionNode still.